src/main/java/co/rsk/federate/signing/hsm/client/HSMClientProtocolFactory.java

```diff
@@ -0,0 +1,30 @@
+package co.rsk.federate.signing.hsm.client;
+
+import co.rsk.federate.config.SignerConfig;
+import co.rsk.federate.rpc.SocketBasedJsonRpcClientProvider;
+import co.rsk.federate.signing.hsm.HSMUnsupportedTypeException;
+
+import java.net.InetSocketAddress;
+
+public class HSMClientProtocolFactory {
+    public static final int DEFAULT_SOCKET_TIMEOUT = 10_000;
+    public static final int DEFAULT_ATTEMPTS = 2;
+    public static final int DEFAULT_INTERVAL = 1000;
+
+    public HSMClientProtocol buildHSMClientProtocolFromConfig(SignerConfig config) throws HSMUnsupportedTypeException {
+        if (!"hsm".equalsIgnoreCase(config.getType())) {
+            throw new HSMUnsupportedTypeException("Config type must be HSM");
+        }
+        String host = config.getConfig().getString("host");
+        int port = config.getConfig().getInt("port");
+        InetSocketAddress hsmAddress = new InetSocketAddress(host, port);
+
+        int socketTimeout = config.getConfig().hasPath("socketTimeout") ? config.getConfig().getInt("socketTimeout") : DEFAULT_SOCKET_TIMEOUT;
+        int maxAttempts = config.getConfig().hasPath("maxAttempts") ? config.getConfig().getInt("maxAttempts") : DEFAULT_ATTEMPTS;
+        int intervalBetweenAttempts = config.getConfig().hasPath("intervalBetweenAttempts") ? config.getConfig().getInt("intervalBetweenAttempts") : DEFAULT_INTERVAL;
```

Comment:
Can these strings be declared in static constants/finals? "socketTimeout", "maxAttempts", "intervalBetweenAttempts", etc. This way they can also be used in the tests knowing that there will be no typo issue, etc.

Reply:
Thanks. Updated